unit-tests/src/main/scala/java/lang/DoubleSuite.scala

```diff
@@ -1,6 +1,18 @@
 package java.lang
 
+import java.lang.Double.{doubleToRawLongBits, longBitsToDouble}
+
 object DoubleSuite extends tests.Suite {
+  test("equals") {
+    assertNot(+0.0 equals -0.0)
+
+    assert(Double.NaN equals Double.NaN)
+
+    val x = Double.NaN
+    val y = longBitsToDouble(doubleToRawLongBits(x) | 1)
+    assert(x equals y)
+  }
```

Comment:
Some interesting values are: `+0.0`, `0.0`, `NaN`, `MaxValue`, `MinValue`, `-42`, `+42`. Would be nice to add more asserts to check that those are equivalent when boxed independently. And not equivalent when you compare the different ones.